In [9]:
import os
os.chdir("D:\\University\\Study Materials\\Capstone\\the-movies-dataset")

In [2]:
import pandas as pd

In [46]:
# Setting Display options 
pd.options.display.max_columns = None

In [10]:
movies = pd.read_csv("tmdb_5000_movies.csv")

credits = pd.read_csv("tmdb_5000_credits.csv")

In [12]:
movies.info(), credits.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4803 entries, 0 to 4802
Data columns (total 20 columns):
budget                  4803 non-null int64
genres                  4803 non-null object
homepage                1712 non-null object
id                      4803 non-null int64
keywords                4803 non-null object
original_language       4803 non-null object
original_title          4803 non-null object
overview                4800 non-null object
popularity              4803 non-null float64
production_companies    4803 non-null object
production_countries    4803 non-null object
release_date            4802 non-null object
revenue                 4803 non-null int64
runtime                 4801 non-null float64
spoken_languages        4803 non-null object
status                  4803 non-null object
tagline                 3959 non-null object
title                   4803 non-null object
vote_average            4803 non-null float64
vote_count              4803 non-null 

(None, None)

In [14]:
movies.head(3)

budget                                             genres  \
0  237000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   
1  300000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...   
2  245000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   

                                       homepage      id  \
0                   http://www.avatarmovie.com/   19995   
1  http://disney.go.com/disneypictures/pirates/     285   
2   http://www.sonypictures.com/movies/spectre/  206647   

                                            keywords original_language  \
0  [{"id": 1463, "name": "culture clash"}, {"id":...                en   
1  [{"id": 270, "name": "ocean"}, {"id": 726, "na...                en   
2  [{"id": 470, "name": "spy"}, {"id": 818, "name...                en   

                             original_title  \
0                                    Avatar   
1  Pirates of the Caribbean: At World's End   
2                                   Spectre   

                                            overview  popularity  \
0  In the 22nd century, a paraplegic Marine is di...  150.437577   
1  Captain Barbossa, long believed to be dead, ha...  139.082615   
2  A cryptic message from Bond’s past sends him o...  107.376788   

                                production_companies  \
0  [{"name": "Ingenious Film Partners", "id": 289...   
1  [{"name": "Walt Disney Pictures", "id": 2}, {"...   
2  [{"name": "Columbia Pictures", "id": 5}, {"nam...   

                                production_countries release_date     revenue  \
0  [{"iso_3166_1": "US", "name": "United States o...   2009-12-10  2787965087   
1  [{"iso_3166_1": "US", "name": "United States o...   2007-05-19   961000000   
2  [{"iso_3166_1": "GB", "name": "United Kingdom"...   2015-10-26   880674609   

   runtime                                   spoken_languages    status  \
0    162.0  [{"iso_639_1": "en", "name": "English"}, {"iso...  Released   
1    169.0           [{"iso_639_1": "en", "name": "English"}]  Released   
2    148.0  [{"iso_639_1": "fr", "name": "Fran\u00e7ais"},...  Released   

                                          tagline  \
0                     Enter the World of Pandora.   
1  At the end of the world, the adventure begins.   
2                           A Plan No One Escapes   

                                      title  vote_average  vote_count  
0                                    Avatar           7.2       11800  
1  Pirates of the Caribbean: At World's End           6.9        4500  
2                                   Spectre           6.3        4466

In [15]:
credits.head(3)

movie_id                                     title  \
0     19995                                    Avatar   
1       285  Pirates of the Caribbean: At World's End   
2    206647                                   Spectre   

                                                cast  \
0  [{"cast_id": 242, "character": "Jake Sully", "...   
1  [{"cast_id": 4, "character": "Captain Jack Spa...   
2  [{"cast_id": 1, "character": "James Bond", "cr...   

                                                crew  
0  [{"credit_id": "52fe48009251416c750aca23", "de...  
1  [{"credit_id": "52fe4232c3a36847f800b579", "de...  
2  [{"credit_id": "54805967c3a36829b5002c41", "de...

In [21]:
master = pd.merge(movies, credits,
                 left_on = "id", right_on = "movie_id",
                 how = "inner")

In [24]:
master.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4803 entries, 0 to 4802
Data columns (total 24 columns):
budget                  4803 non-null int64
genres                  4803 non-null object
homepage                1712 non-null object
id                      4803 non-null int64
keywords                4803 non-null object
original_language       4803 non-null object
original_title          4803 non-null object
overview                4800 non-null object
popularity              4803 non-null float64
production_companies    4803 non-null object
production_countries    4803 non-null object
release_date            4802 non-null object
revenue                 4803 non-null int64
runtime                 4801 non-null float64
spoken_languages        4803 non-null object
status                  4803 non-null object
tagline                 3959 non-null object
title_x                 4803 non-null object
vote_average            4803 non-null float64
vote_count              4803 non-null 

(None, (4803, 24))

In [26]:
master.head(3)

budget                                             genres  \
0  237000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   
1  300000000  [{"id": 12, "name": "Adventure"}, {"id": 14, "...   
2  245000000  [{"id": 28, "name": "Action"}, {"id": 12, "nam...   

                                       homepage      id  \
0                   http://www.avatarmovie.com/   19995   
1  http://disney.go.com/disneypictures/pirates/     285   
2   http://www.sonypictures.com/movies/spectre/  206647   

                                            keywords original_language  \
0  [{"id": 1463, "name": "culture clash"}, {"id":...                en   
1  [{"id": 270, "name": "ocean"}, {"id": 726, "na...                en   
2  [{"id": 470, "name": "spy"}, {"id": 818, "name...                en   

                             original_title  \
0                                    Avatar   
1  Pirates of the Caribbean: At World's End   
2                                   Spectre   

                                            overview  popularity  \
0  In the 22nd century, a paraplegic Marine is di...  150.437577   
1  Captain Barbossa, long believed to be dead, ha...  139.082615   
2  A cryptic message from Bond’s past sends him o...  107.376788   

                                production_companies  ...  \
0  [{"name": "Ingenious Film Partners", "id": 289...  ...   
1  [{"name": "Walt Disney Pictures", "id": 2}, {"...  ...   
2  [{"name": "Columbia Pictures", "id": 5}, {"nam...  ...   

                                    spoken_languages    status  \
0  [{"iso_639_1": "en", "name": "English"}, {"iso...  Released   
1           [{"iso_639_1": "en", "name": "English"}]  Released   
2  [{"iso_639_1": "fr", "name": "Fran\u00e7ais"},...  Released   

                                          tagline  \
0                     Enter the World of Pandora.   
1  At the end of the world, the adventure begins.   
2                           A Plan No One Escapes   

                                    title_x vote_average vote_count movie_id  \
0                                    Avatar          7.2      11800    19995   
1  Pirates of the Caribbean: At World's End          6.9       4500      285   
2                                   Spectre          6.3       4466   206647   

                                    title_y  \
0                                    Avatar   
1  Pirates of the Caribbean: At World's End   
2                                   Spectre   

                                                cast  \
0  [{"cast_id": 242, "character": "Jake Sully", "...   
1  [{"cast_id": 4, "character": "Captain Jack Spa...   
2  [{"cast_id": 1, "character": "James Bond", "cr...   

                                                crew  
0  [{"credit_id": "52fe48009251416c750aca23", "de...  
1  [{"credit_id": "52fe4232c3a36847f800b579", "de...  
2  [{"credit_id": "54805967c3a36829b5002c41", "de...  

[3 rows x 24 columns]

In [32]:
master.id.nunique(), master.original_title.nunique()  #2 duplicates in original title

(4803, 4801)

In [36]:
from collections import Counter

d =  Counter(master.original_title)
res = [k for k, v in d.items() if v > 1]
print(res)

['Batman', 'Out of the Blue']


In [47]:
master[master.original_title == "Batman"] #not duplicate

budget                                             genres homepage  \
1359  35000000  [{"id": 14, "name": "Fantasy"}, {"id": 28, "na...      NaN   
4267   1377800  [{"id": 10751, "name": "Family"}, {"id": 12, "...      NaN   

        id                                           keywords  \
1359   268  [{"id": 848, "name": "double life"}, {"id": 84...   
4267  2661  [{"id": 339, "name": "submarine"}, {"id": 849,...   

     original_language original_title  \
1359                en         Batman   
4267                en         Batman   

                                               overview  popularity  \
1359  The Dark Knight of Gotham City begins his war ...   44.104469   
4267  The Dynamic Duo faces four super-villains who ...    9.815394   

                                   production_companies  \
1359  [{"name": "PolyGram Filmed Entertainment", "id...   
4267  [{"name": "Twentieth Century Fox Film Corporat...   

                                   production_countries release_date  \
1359  [{"iso_3166_1": "GB", "name": "United Kingdom"...   1989-06-23   
4267  [{"iso_3166_1": "US", "name": "United States o...   1966-07-30   

        revenue  runtime                                   spoken_languages  \
1359  411348924    126.0  [{"iso_639_1": "en", "name": "English"}, {"iso...   
4267          0    105.0           [{"iso_639_1": "en", "name": "English"}]   

        status                                            tagline title_x  \
1359  Released  Have you ever danced with the devil in the pal...  Batman   
4267  Released           He's Here Big As Life In A Real Bat-Epic  Batman   

      vote_average  vote_count  movie_id title_y  \
1359           7.0        2096       268  Batman   
4267           6.1         203      2661  Batman   

                                                   cast  \
1359  [{"cast_id": 5, "character": "Jack Napier/The ...   
4267  [{"cast_id": 17, "character": "Batman / Bruce ...   

                                                   crew  
1359  [{"credit_id": "52fe422fc3a36847f800aa4b", "de...  
4267  [{"credit_id": "52fe4363c3a36847f80509a7", "de...

In [48]:
master[master.original_title == "Out of the Blue"] #not duplicate

budget                         genres homepage     id  \
3647       0  [{"id": 18, "name": "Drama"}]      NaN  39269   
3693       0  [{"id": 18, "name": "Drama"}]      NaN  10844   

                                               keywords original_language  \
3647  [{"id": 4470, "name": "punk"}, {"id": 10183, "...                en   
3693  [{"id": 2658, "name": "new zealand"}, {"id": 3...                en   

       original_title                                           overview  \
3647  Out of the Blue  Dennis Hopper is a hard-drinking truck driver ...   
3693  Out of the Blue  Ordinary people find extraordinary courage in ...   

      popularity                      production_companies  \
3647    0.679351  [{"name": "Robson Street", "id": 71953}]   
3693    0.706355                                        []   

                               production_countries release_date  revenue  \
3647       [{"iso_3166_1": "CA", "name": "Canada"}]   1980-05-01        0   
3693  [{"iso_3166_1": "NZ", "name": "New Zealand"}]   2006-10-12        0   

      runtime                          spoken_languages    status  \
3647     94.0  [{"iso_639_1": "en", "name": "English"}]  Released   
3693    103.0  [{"iso_639_1": "en", "name": "English"}]  Released   

                                                tagline          title_x  \
3647  A harrowing drama from the director of Easy Rider  Out of the Blue   
3693            The true story of a small town massacre  Out of the Blue   

      vote_average  vote_count  movie_id          title_y  \
3647           6.5          17     39269  Out of the Blue   
3693           5.9          18     10844  Out of the Blue   

                                                   cast  \
3647  [{"cast_id": 2, "character": "Don", "credit_id...   
3693  [{"cast_id": 12, "character": "Nick Harvey", "...   

                                                   crew  
3647  [{"credit_id": "52fe47099251416c9106826f", "de...  
3693  [{"credit_id": "52fe43c19251416c7501cceb", "de...

In [45]:
#missing values
master.isna().sum()

budget                     0
genres                     0
homepage                3091
id                         0
keywords                   0
original_language          0
original_title             0
overview                   3
popularity                 0
production_companies       0
production_countries       0
release_date               1
revenue                    0
runtime                    2
spoken_languages           0
status                     0
tagline                  844
title_x                    0
vote_average               0
vote_count                 0
movie_id                   0
title_y                    0
cast                       0
crew                       0
dtype: int64

4803 unique movies available
some variables contains missing values